packages/dds/sequence/src/localValues.ts

```diff
@@ -95,18 +95,15 @@ export class PlainLocalValue implements ILocalValue {
 /**
  * Manages a contained value type.
  *
- * @privateRemarks
- * TODO: Should maybe be a generic
- *
  * @alpha
  */
-export class ValueTypeLocalValue implements ILocalValue {
+export class ValueTypeLocalValue<T> implements ILocalValue<T> {
```

Comment:
it might actually be better to start at the bottom and collapse the abstractions upward by inclining specific types, and trimming anything uneeded

Reply:
yeah I think that's right.

Reply:
plain/shared support and definitions should be ripped out now. I left an assert in on local creation, though honestly it's probably safe to take that out as well.

We could rip out the value type / serialization logic separation abstraction as well, but it might be nicer to do that in a different PR since it's going to cause format changes (I'm assuming as part of that change we'll want to remove the unnecessary "type" field that always points to an interval collection).

Reply:
either way is fine with me.

Reply:
Authoring-MVP3 has snapshot contents containing Plain value types with handles. See e.g. near L341 of packages\test\snapshots\content\snapshotTestContent\Authoring-MVP3\current_snapshots\snapshot_12483_0.json

                            "entries": [
                              {
                                "mode": "100644",
                                "path": "header",
                                "type": "Blob",
                                "value": {
                                  "contents": "{\"intervalCollections\":{\"type\":\"Plain\",\"value\":{\"type\":\"__fluid_handle__\",\"url\":\"/defaultComponent/347f0495-5605-4622-bcf1-0571a2bc2178\"}}}",
                                  "encoding": "utf-8"
                                }
                              },


So ripping out this support causes some test failures there. Is there a reason to believe that doc isn't legitimate?

Reply:
this doesn't look legitimate to me the fact that there is a handle under the intervalCollections key is especially odd. it shouldn't be possible to create or use document in this state.

Reply:
Added some code which allows loading documents with plain and shared value types under their interval collections, but silently drops them. This is correct behavior as long as any documents that were created this way didn't use the getIntervalCollections API. If they did, we're risking dropping any intervals that were added. Though, as you said, it doesn't look like the current version of the code even supports those intervals anyway. It seems highly unlikely anyone actually expects this to work. I think opening such old documents without crashing is sufficient.